Scripts/Set-CosmosDbAccountComponents.ps1

```diff
@@ -78,25 +78,14 @@ if (!(Get-Module CosmosDB | Where-Object { $_.Version.ToString() -eq "2.1.3.528"
 }
 
 Write-Verbose -Message "Searching for existing account"
-$AzureRmMajorVersion = (((Get-Module AzureRM -ListAvailable | Sort-Object { $_.Version.Major } -Descending).Version.Major))[0]
-Write-Verbose "AzureRm Major Version: $AzureRmMajorVersion"
-if ($AzureRmMajorVersion -gt 5) {
-    $GetCosmosDbAccountParameters = @{
-        Name              = $CosmosDbAccountName
-        ResourceGroupName = $ResourceGroupName
-        ExpandProperties  = $true
-        ResourceType      = "Microsoft.DocumentDB/databaseAccounts"
-    }
-    $ExistingAccount = Get-AzureRmResource @GetCosmosDbAccountParameters
-}
-else {
-    $GetCosmosDbAccountParameters = @{
-        ResourceType      = "Microsoft.DocumentDb/databaseAccounts"
-        ResourceGroupName = $ResourceGroupName
-        ResourceName      = $CosmosDbAccountName
-    }
-    $ExistingAccount = Get-AzureRmResource @GetCosmosDbAccountParameters
+
+$GetCosmosDbAccountParameters = @{
+    Name              = $CosmosDbAccountName
+    ResourceGroupName = $ResourceGroupName
+    ExpandProperties  = $true
+    ResourceType      = "Microsoft.DocumentDB/databaseAccounts"
 }
+$ExistingAccount = Get-AzResource @GetCosmosDbAccountParameters
 
 if (!$ExistingAccount -or $ExistingAccount.Properties.provisioningState -ne "Succeeded") {
```

Comment:
if you haven't already, it's worth checking if Get-AzResource does return .Properties.provisioningState if you haven't already!